pkg/metaserver/application_test.go

```diff
@@ -0,0 +1,409 @@
+package metaserver
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"testing"
+
+	"github.com/kubeedge/beehive/pkg/core/model"
+	"github.com/stretchr/testify/assert"
```

Comment:
Is this a new import lib? Is there an existing lib that can be used as its substitute? If not, we ned to add this lib add go.mod file and the run `make verify` to update the vendor.

Reply:
this is a new import lib (we have `github.com/stretchr/testify` in go.mod ) . this is an assertion lib that makes the code shorter and much more readable. yes, it can be removed and I can simply just use the testing package. But I think using this lib will be better for reasons I mentioned ([there also somebody who created an issue about it](https://github.com/kubeedge/kubeedge/issues/4403), as he said "This removes much boilerplate from the tests, and improves speed and quality of understanding and modifying them." ) . my suggestion would be using testify, you can tell me which one do you prefer and I will use that in coming unit tests as well :)

Reply:
I just saw this lib being used in these [tests](https://github.com/kubeedge/kubeedge/pull/2793/files)

Reply:
I think `github.com/stretchr/testify` is fine. Go ahead! Or we can use it to do a full replacement in all unit tests in another PR.  
I have find `github.com/stretchr/testify` in go.mod. If we need to use it, we may need to remove `//indirect` and run `make verify` to update the related vendor.

Reply:
Shelley, I did as you said, I removed `//indirect` and then run `make verify`. But it gives error at :
```
        /go/pkg/mod/sigs.k8s.io/controller-tools@v0.6.2/cmd/controller-gen/main.go:200 +0x2f6
make: *** [Makefile:96: verify-crds] Error 2
```

Reply:
`make verify` has succeed in CI. I guess the error in your environment is caused by golang version?

Reply:
my go version is `1.22.3`

Reply:
one test case still failing, I don't understand why, I didn't even change the code for this to fail at 

```
 [FAILED] in [It] - /home/runner/work/kubeedge/kubeedge/edge/test/integration/utils/helpers/helpers.go:265
```

I have simply added tests.

Reply:
This error should be unrelated with you. I will retry the test : )

Reply:
edit: now all tests are passing.